### PR TITLE
Bring back .optional how it originally was

### PR DESCRIPTION
**Pull request by tonyd256**

### CODE REVIEW

File: Argo/Types/Decoded/Decoded.swift [hunk 1]
```diff
@@ -32,6 +32,21 @@ public extension Decoded {
...
 public extension Decoded {
   /**
+    Convert a `Decoded` type into a `Decoded` `Optional` type.
+
+    This is useful for when a decode operation should be allowed to fail, such
+    as when decoding an optional property.
+
+    - parameter x: A `Decoded` type
+
+    - returns: The `Decoded` type with a `.TypeMismatch` failure converted to
```
Comment (gfontenot) [inline Argo/Types/Decoded/Decoded.swift]: Is this accurate? This is just forcing _any_ `Decoded` to be `Success` with an optional value now, right? (also, these are lowercased now)

**Comment by gfontenot** [comment]
@tonyd256 do you have time to fix up the documentation and get this into master? I'd like to ship this in the release alongside #425 and #424

**Comment by tonyd256** [comment]
@gfontenot I could friday at the earliest

**Comment by tonyd256** [comment]
@gfontenot added a test case for the optional function ... want to check that out then I'll squash and merge

**Comment by gfontenot** [comment]
Dig the test, still think the documentation needs to be fixed before we merge this.

**Comment by tonyd256** [comment]
oh right .. forgot about the docs ... on it

File: Argo/Types/Decoded/Decoded.swift [hunk 2]
```diff
@@ -32,6 +32,20 @@ public extension Decoded {
...
 public extension Decoded {
   /**
+    Convert a `Decoded` type into a `Decoded` `Optional` type.
+
+    This is useful for when a decode operation should be allowed to fail, such
+    as when decoding an optional property.
+
+    - parameter x: A `Decoded` type
+
+    - returns: The `Decoded` type with any failure converted to `.Success(.None)`
```
Comment (gfontenot) [inline Argo/Types/Decoded/Decoded.swift]: Lowercase `.success` and `.none`?

**Pull request merged**